src/lucky/log_handler.cr

```diff
@@ -2,9 +2,13 @@ require "colorize"
 
 class Lucky::LogHandler
   include HTTP::Handler
+  alias LogProc = ::Proc(Lucky::LogHandler, HTTP::Server::Context, Time, Time::Span, String)
 
   Habitat.create do
     setting show_timestamps : Bool
+    setting request_formatter : LogProc = ->(handler : Lucky::LogHandler, context : HTTP::Server::Context, time : Time, elapsed : Time::Span) {
+      "#{context.request.method} #{handler.colored_status_code(context.response.status_code)} #{context.request.resource}#{handler.timestamp(time)} (#{handler.elapsed_text(elapsed)})"
+    }
```

Comment:
How about instead of using a Proc, there would be a class in here called `LogFormatter`

Something like this:

```
abstract class LogFormatter
  abstract def format(context, time, elapsed) : String

  # Include elapsed_text, colored_status_code, etc.
end
```

Then the default would be:

```
class DefaultLogFormatter < LogFormatter
  def log_request(context, time, elapsed)
    "#{context.request.method} #{handler.colored_status_code(context.response.status_code)} #{context.request.resource}#{handler.timestamp(time)} (#{handler.elapsed_text(elapsed)})"
  end
end

setting log_formatter : LogFormatter = DefaultLogFormatter.new
```

Does that make sense?

Reply:
ah. nice. That might be nicer. I'll try that out.